### PR TITLE
Remove trailing dash on header IDs when using LifeCycle component

### PR DESCRIPTION
**Pull request by JKarlavige**
[Asana task](https://app.asana.com/0/1200099998847559/1206616034730932/f)

### Issue
If the `LifeCycle` component is used on a header, Docusaurus adds a trailing `-` at the end of the header id attribute. If you click to copy a header using the LifeCycle component, you will see the trailing dash included to the url in your clipboard. 

While this works, it may cause confusion when the team is sharing links to specific sections.

### Resolution
Check if header id has a trailing dash, and if so, remove trailing dash on initial load of page.

### Checking header with single LifeCycle status

On this page on the live site, a LifeCycle component is used on the `Keep on latest version` header. This causes a trailing dash to be added to the `id` for this header element:
https://docs.getdbt.com/docs/dbt-versions/upgrade-dbt-version-in-cloud

In the preview:
- verify the trailing slash is removed from the `Keep on latest version` id attribute
- verify you can click the header to copy it, and that the trailing dash is removed
- verify other headers have correct id set and the click-to-copy function works as expected

https://docs-getdbt-com-git-remove-trailing-dash-dbt-labs.vercel.app/docs/dbt-versions/upgrade-dbt-version-in-cloud

### Checking header with multiple LifeCycle statuses set

On this page, multiple statuses are used in the `Trigger on job completion` header, causing multiple trailing dashes:
https://docs.getdbt.com/docs/deploy/deploy-jobs#trigger-on-job-completion--

Verify all trailing dashes are removed within the preview
https://docs-getdbt-com-git-remove-trailing-dash-dbt-labs.vercel.app/docs/deploy/deploy-jobs
## Notes

I have a couple console logs in the preview to verify the script is detecting and removing the trailing slashes.

## TODO Before Merge
- [ ] Remove console logs

### CODE REVIEW

File: website/static/js/headerLinkCopy.js [hunk 1]
```diff
@@ -1,58 +1,84 @@
...
+// separating function from eventlistener to understand they are two separate things
+function copyHeader () {
+  const headers = document.querySelectorAll("h2.anchor, h3.anchor, .expandable-anchor.anchor");
 
-headers.forEach((header) => {
+  headers.forEach((header) => {
     header.style.cursor = "pointer";
+
+    // Remove trailing `-` if exists
+    header = removeTrailingDashes(header)
```
Comment (JKarlavige) [inline website/static/js/headerLinkCopy.js:17]: New code added here

File: website/static/js/headerLinkCopy.js [hunk 2]
```diff
@@ -1,58 +1,84 @@
...
+    // Remove `-` from end of ID string
+    updatedHeader.id = thisId?.substring(0, thisId?.length - 1);
+    console.log(`Header ID after update: ${updatedHeader.id}`);
+
+    // Recursively run function to check for another trailing slash
+    removeTrailingDashes(updatedHeader);
+  }
+
+  return updatedHeader;
+}
```
Comment (JKarlavige) [inline website/static/js/headerLinkCopy.js]: New code added here

**Review by mirnawong1 — APPROVED** [review]
i've tested on my end and it looks great to me, working beautifully! approved it but will let others test it on their end too! thank you so much @JKarlavige! this --> `header.id = thisId?.substring(0, thisId?.length - 1)` is very cool to learn!

**Review by john-rock — APPROVED** [review]
lgtm! Verified trailing dash is removed and click-to-copy functionality works on titles both with and without the lifecycle component.

File: website/static/js/headerLinkCopy.js (hunk 1 above)
Comment (nghi-ly) [reply website/static/js/headerLinkCopy.js:17]: neat! love the comments!

**Review by nghi-ly — APPROVED** [review]
tysm for fixing this bug!

**Pull request merged**